specification/appendix-d-observability.md

```diff
@@ -0,0 +1,50 @@
+| ------------------------ | --------------------------------------- | ---------------------------------------------------- |
+| `flag key`               | `feature_flag.key`                      | See: [flag key](./glossary.md#flag-key)              |
+| `error code`             | `error.type`                            | See: [error code](./types.md#error-code)             |
+| `variant`, `value`       | `feature_flag.variant`                  | See: [variant](./glossary.md#variant)                |
+| `error message`          | `feature_flag.evaluation.error.message` | An error message associated with a failed evaluation |
+| `reason`                 | `feature_flag.evaluation.reason`        | See: [reason](./types.md#reason)                     |
+
+> [!NOTE]  
+> If the provider in question doesn't supply a `variant`, the a string representation of the `value` is used.
+
+### Flag Metadata
+
+The following describes how keys in [flag metadata](types.md#flag-metadata) are mapped to feature flag log records:
+
+| Flag Metadata Key | Log Record Attribute      | Notes                                                                                                                                                                                           |
+| ----------------- | ------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `contextId`       | `feature_flag.context.id` | The context identifier returned in the flag metadata uniquely identifies the subject of the flag evaluation. If not available, the [targeting key](./glossary.md#targeting-key) should be used. |
+| `flagSetId`       | `feature_flag.set.id`     | A logical identifier for the [flag set](./glossary.md#flag-set).                                                                                                                                |
+| `flagSetVersion`  | `feature_flag.version`    | A version string (format unspecified) for the [flag set](./glossary.md#flag-set).                                                                                                               |
+
+> [!NOTE]  
+> Keys in flag metadata use the "camelCase" casing convention, while the OpenTelemetry standard uses a namespaced "snake_case" convention.
+
+### Provider Metadata
+
+| Provider Metadata Field | Log Record Attribute         | Notes                                                                                            |
+| ----------------------- | ---------------------------- | ------------------------------------------------------------------------------------------------ |
+| `name`                  | `feature_flag.provider_name` | The name of the provider as defined in the `provider metadata`, available in the `hook context`. |
```

Comment:
I wonder if we should add a note that it's recommended to keep provider names consistent across provider implementations.

Reply:
I don't think it's worth mentioning. I guess providers can decide for themselves if they want the name to be consistent.

Reply:
I think so too @toddbaert, it might be interesting for some consumers to see the provider implementations instead of a common name. But I guess @beeme1mr's case is the more common.

Reply:
I'm also with @toddbaert regarding a provider name. But I can imagine your point @beeme1mr of having an identifiying property for post processing. Maybe there should also be a `provider_id` property we could propose in otel?